File: Python/Course.py
from Dosen import Dosen
from Mahasiswa import Mahasiswa

class Course():
    def __init__(self, nama_matakuliah="-", listDosen=[], listMahasiswa=[]):
        self.__nama_matakuliah = nama_matakuliah
        self.__listDosen = listDosen
        self.__listMahasiswa = listMahasiswa
    
    def setNama_matakuliah(self, nama_matakuliah):
        self.__nama_matakuliah = nama_matakuliah

    def getNama_matakuliah(self):
        return self.__nama_matakuliah

    def setListDosen(self, listDosen):
        self.__listDosen = listDosen

    def getListDosen(self):
        return self.__listDosen
    
    def addListDosen(self, dosen):
        self.__listDosen.append(dosen)
    
    def getStrListDosen(self):
        return ', '.join(str(e.getNama()) for e in self.getListDosen() )
	
    def setListMahasiswa(self, listMahasiswa):
        self.__listMahasiswa = listMahasiswa
	
    def getListMahasiswa(self):
        return self.__listMahasiswa
    
    def addListMahasiswa(self, mahasiswa):
        self.__listMahasiswa.append(mahasiswa)
    
    def getStrListMahasiswa(self):
        return ', '.join(str(e.getNama()) for e in self.getListMahasiswa() )

File: cpp/Course.cpp
// course memiliki list objek dosen dan mahasiswa

class Course
{
	private:
		string nama_matakuliah;
		vector<Dosen> listDosen;
		vector<Mahasiswa> listMahasiswa;
	public:
		Course(){
			this->nama_matakuliah = "-";
		}
		Course(string nama_matakuliah, vector<Dosen> listDosen, vector<Mahasiswa> listMahasiswa){
			this->nama_matakuliah = nama_matakuliah;
			this->listDosen = listDosen;
			this->listMahasiswa = listMahasiswa;
		}
		void setNama_matakuliah(string nama_matakuliah){
			this->nama_matakuliah = nama_matakuliah;
		}
		string getNama_matakuliah(){
			return this->nama_matakuliah;
		}
		void setListDosen(vector<Dosen> listDosen){
			this->listDosen = listDosen;
		}
		vector<Dosen> getListDosen(){
			return this->listDosen;
		}
		void addListDosen(Dosen dosen){
			this->listDosen.push_back(dosen);
		}
		string getStrListDosen(){		//mengembalikan nama nama dosen pada listDosen
			string dosen;
			for(vector<Dosen>::iterator ptr=listDosen.begin(); ptr<listDosen.end(); ptr++){
				dosen.append((*ptr).getNama());		//semua nama dosen disimpan pada string dosen
				if(ptr+1 != listDosen.end()){
					dosen.append(", ");
				}
			}
			return dosen;
		}
		void setListMahasiswa(vector<Mahasiswa> listMahasiswa){
			this->listMahasiswa = listMahasiswa;
		}
		vector<Mahasiswa> getListMahasiswa(){
			return this->listMahasiswa;
		}
		void addListMahasiswa(Mahasiswa mahasiswa){
			this->listMahasiswa.push_back(mahasiswa);
		}
		string getStrListMahasiswa(){		//mengembalikan nama nama mahasiswa pada listDosen
			string mhs;
			for(vector<Mahasiswa>::iterator ptr=listMahasiswa.begin(); ptr<listMahasiswa.end(); ptr++){
				mhs.append((*ptr).getNama());		//semua nama mahasiswa disimpan pada string mhs
				if(ptr+1 != listMahasiswa.end()){
					mhs.append(", ");
				}
			}
			return mhs;
		}
};
